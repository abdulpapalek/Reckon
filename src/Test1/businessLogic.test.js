/* eslint-disable */
const BusinessLogic = require("./businessLogic");
const Api = require("./api");

describe("BusinessLogic", () => {  
    afterEach(() => {
      // cleaning up the mess left behind the previous test
      jest.clearAllMocks();
    });
  
    test("Success: Should return the correct list inside html template", async () => {
        Api.getRangeInfo = jest.fn(async () => Promise.resolve({
            lower: 0,
            upper: 100
          }));
        Api.getDivisorInfo = jest.fn(async () => Promise.resolve({
              outputDetails: [
                  {
                      divisor: 3,
                      output: "Peppa"
                  },
                  {
                      divisor: 5,
                      output: "Pig"
                  }
              ]
          }));
        expect(await BusinessLogic.buildList()).toMatchSnapshot();;
      });
});