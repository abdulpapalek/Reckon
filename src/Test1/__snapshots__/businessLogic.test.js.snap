// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessLogic Success: Should return the correct list inside html template 1`] = `"<tr><td>0: Peppa</td></tr><tr><td>1:</td></tr><tr><td>2:</td></tr><tr><td>3: Peppa</td></tr><tr><td>4:</td></tr><tr><td>5: Pig</td></tr><tr><td>6: Peppa</td></tr><tr><td>7:</td></tr><tr><td>8:</td></tr><tr><td>9: Peppa</td></tr><tr><td>10: Pig</td></tr><tr><td>11:</td></tr><tr><td>12: Peppa</td></tr><tr><td>13:</td></tr><tr><td>14:</td></tr><tr><td>15: Peppa</td></tr><tr><td>16:</td></tr><tr><td>17:</td></tr><tr><td>18: Peppa</td></tr><tr><td>19:</td></tr><tr><td>20: Pig</td></tr><tr><td>21: Peppa</td></tr><tr><td>22:</td></tr><tr><td>23:</td></tr><tr><td>24: Peppa</td></tr><tr><td>25: Pig</td></tr><tr><td>26:</td></tr><tr><td>27: Peppa</td></tr><tr><td>28:</td></tr><tr><td>29:</td></tr><tr><td>30: Peppa</td></tr><tr><td>31:</td></tr><tr><td>32:</td></tr><tr><td>33: Peppa</td></tr><tr><td>34:</td></tr><tr><td>35: Pig</td></tr><tr><td>36: Peppa</td></tr><tr><td>37:</td></tr><tr><td>38:</td></tr><tr><td>39: Peppa</td></tr><tr><td>40: Pig</td></tr><tr><td>41:</td></tr><tr><td>42: Peppa</td></tr><tr><td>43:</td></tr><tr><td>44:</td></tr><tr><td>45: Peppa</td></tr><tr><td>46:</td></tr><tr><td>47:</td></tr><tr><td>48: Peppa</td></tr><tr><td>49:</td></tr><tr><td>50: Pig</td></tr><tr><td>51: Peppa</td></tr><tr><td>52:</td></tr><tr><td>53:</td></tr><tr><td>54: Peppa</td></tr><tr><td>55: Pig</td></tr><tr><td>56:</td></tr><tr><td>57: Peppa</td></tr><tr><td>58:</td></tr><tr><td>59:</td></tr><tr><td>60: Peppa</td></tr><tr><td>61:</td></tr><tr><td>62:</td></tr><tr><td>63: Peppa</td></tr><tr><td>64:</td></tr><tr><td>65: Pig</td></tr><tr><td>66: Peppa</td></tr><tr><td>67:</td></tr><tr><td>68:</td></tr><tr><td>69: Peppa</td></tr><tr><td>70: Pig</td></tr><tr><td>71:</td></tr><tr><td>72: Peppa</td></tr><tr><td>73:</td></tr><tr><td>74:</td></tr><tr><td>75: Peppa</td></tr><tr><td>76:</td></tr><tr><td>77:</td></tr><tr><td>78: Peppa</td></tr><tr><td>79:</td></tr><tr><td>80: Pig</td></tr><tr><td>81: Peppa</td></tr><tr><td>82:</td></tr><tr><td>83:</td></tr><tr><td>84: Peppa</td></tr><tr><td>85: Pig</td></tr><tr><td>86:</td></tr><tr><td>87: Peppa</td></tr><tr><td>88:</td></tr><tr><td>89:</td></tr><tr><td>90: Peppa</td></tr><tr><td>91:</td></tr><tr><td>92:</td></tr><tr><td>93: Peppa</td></tr><tr><td>94:</td></tr><tr><td>95: Pig</td></tr><tr><td>96: Peppa</td></tr><tr><td>97:</td></tr><tr><td>98:</td></tr><tr><td>99: Peppa</td></tr><tr><td>100: Pig</td></tr>"`;
